prov/shm/src/smr_progress.c

```diff
@@ -106,6 +106,15 @@ static void smr_progress_resp(struct smr_ep *ep)
 		}
 
 		if (pending->map_name) {
+			// read data out of mmapped shm file
+			if (pending->cmd.msg.hdr.op == ofi_op_read_req) {
+				ret = ofi_copy_to_iov(pending->iov,
+						pending->iov_count, 0,
+						pending->map_ptr,
+						pending->cmd.msg.hdr.size);
+				munmap(pending->map_ptr, pending->cmd.msg.hdr.size);
+			}
+
```

Comment:
Again, see comment about looking at #5682 for generalization of smr_progress_resp to increment the cmd_cnt. I think we could pretty easily add a switch on the pending->cmd.msg.hdr.op to either just increment, copy from the map_ptr or copy from the tx buf depending on the op type.

Reply:
I saw your change in #5682 for generalization of smr_progress_resp to increment the cmd_cnt, and I agree with your idea of adding a switch. 

Regarding to the changes (formatting and generalization of smr_progress_resp), I prefer rebasing and updating once your PR #5682 gets merged, in case of you want to do more changes. Hope it's okay to you.

Reply:
Yeah, that sounds good to me. Thank you!